File: engine/News/htmlSearcher.py
from multiprocessing.pool import Pool
import re

def init(urlInfoArg, minNumOccurencesArg):
    global urlInfo
    urlInfo = urlInfoArg
    global minNumOccurences
    minNumOccurences = minNumOccurencesArg


def pre_search_html(html):
    return search_html(html, urlInfo, minNumOccurences)


def parallel_html_search(htmls, urlInfo, numPools, minNumOccurences):
    with Pool(numPools, init, (urlInfo, minNumOccurences,)) as pool:
        return pool.map(pre_search_html, htmls)


def sequential_html_search(htmls, urlInfo, minNumOccurences):
    return [search_html(html, urlInfo, minNumOccurences) for html in htmls]


def search_html(html, urlInfo, minNumOccurences):
    for articleClass in urlInfo.articleClasses:
        if(len(re.findall(articleClass, html)) >= minNumOccurences): return True
    return False

File: gui/linkers/modules/utils/htmlSearcher.js
function sequential_html_search(htmls, newsUrlInfo, minNumOccurences){
    let articleClasses = Object.keys(newsUrlInfo.articlesInfo)
    let length = articleClasses.length

    return htmls.map(html => search_html(html, articleClasses, length, minNumOccurences))
}

function search_html(html, articleClasses, articleClassesLength = null, minNumOccurences=2){
    if(html){
        if(!articleClassesLength) articleClassesLength = articleClasses.length
        
        for(let i=0; i<articleClassesLength; i++){
            let matches = html.match(new RegExp(`"${articleClasses[i]}"`, "g"))

            if(matches && matches.length >= minNumOccurences) return true
        }
    }

    return false
}

export {sequential_html_search, search_html}